### PR TITLE
feat: readme & examples

### PR DESCRIPTION
**Pull request by caarlos0**
- better readme
- example printing all colors for all variants

### CODE REVIEW

**Comment by caarlos0** [comment]
reminder: squash & merge to not have too many gifs in history :P

File: frappe.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ type frappe struct{}
 
 var _ Theme = frappe{}
 
-func (frappe) Name() string { return "frappe" }
+func (frappe) Name() string { return "Frappe" }
```
Comment (sgoudham) [inline frappe.go]: We should probably keep consistency with the Rust equivalent here: https://github.com/catppuccin/rust/blob/main/src/flavour.rs#L28-L33

```suggestion
func (frappe) Name() string { return "frappe" }
```

File: latte.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ type latte struct{}
 
 var _ Theme = latte{}
 
-func (latte) Name() string { return "latte" }
+func (latte) Name() string { return "Latte" }
```
Comment (sgoudham) [inline latte.go]: Same comment as above.
```suggestion
func (latte) Name() string { return "latte" }
```

File: macchiato.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ type macchiato struct{}
 
 var _ Theme = macchiato{}
 
-func (macchiato) Name() string { return "macchiato" }
+func (macchiato) Name() string { return "Macchiato" }
```
Comment (sgoudham) [inline macchiato.go]: Same comment as above.
```suggestion
func (macchiato) Name() string { return "macchiato" }
```

File: mocha.go [hunk 4]
```diff
@@ -4,7 +4,7 @@ type mocha struct{}
 
 var _ Theme = mocha{}
 
-func (mocha) Name() string { return "mocha" }
+func (mocha) Name() string { return "Mocha" }
```
Comment (sgoudham) [inline mocha.go]: Same comment as above.
```suggestion
func (mocha) Name() string { return "mocha" }
```

**Review by sgoudham — CHANGES_REQUESTED** [review]
Looking very snazzy! I just had a few comments about the naming. 

Let me know if they're valid or not. I'm not willing to die on that hill :P

**Comment by caarlos0** [comment]
all done!

**Review by sgoudham — COMMENTED** [review]
Thanks for merging!

**Pull request merged**